resources/js/components/dashboard.js

```diff
@@ -0,0 +1,216 @@
+                        widget.class = item.gridstackNode.class;
+                    }
+                    // in case something is selected from the list
+                    if (item.gridstackNode.component_name !== undefined) {
+                        widget.component_name =
+                            item.gridstackNode.component_name;
+                    }
+                    newSnapshot.push(widget);
+                } else {
+                    // new widget on the screen
+                    newSnapshot.push({
+                        id: item.gridstackNode.id,
+                        height: item.gridstackNode.h,
+                        width: item.gridstackNode.w,
+                        order_column: item.gridstackNode.x,
+                        order_row: item.gridstackNode.y,
+                        component_name: item.gridstackNode.component_name
+                    });
+                }
+            });
+            // sync property
+            await $wire.syncWidgets(newSnapshot);
+        },
+        async syncGridOnDelete() {
+            const snapshot = $wire.widgets;
+            const onScreen = this.grid.getGridItems();
+            const newSnapshot = [];
+            onScreen.forEach((item) => {
+                const widget = snapshot.find(
+                    (w) => w.id.toString() === item.gridstackNode.id.toString()
+                );
+                // remove from snapshot if not on the screen and recalculate x,y coordinates
+                if (item.style.display !== 'none' && widget !== undefined) {
+                    widget.height = item.gridstackNode.h;
+                    widget.width = item.gridstackNode.w;
+                    widget.order_column = item.gridstackNode.x;
+                    widget.order_row = item.gridstackNode.y;
+                    newSnapshot.push(widget);
+                }
+            });
+            // sync property
+            await $wire.syncWidgets(newSnapshot);
+        },
+        async save() {
+            const snapshot = $wire.widgets;
+            console.log('snapshot', snapshot);
+            console.log('onScreen', this.grid.getGridItems());
+            const onScreen = this.grid.getGridItems();
+            const newSnapshot = [];
+            // update x,y coordinates on save
+            onScreen.forEach((item) => {
+                const widget = snapshot.find(
+                    (w) => w.id.toString() === item.gridstackNode.id.toString()
+                );
+                if (widget !== undefined) {
+                    widget.height = item.gridstackNode.h;
+                    widget.width = item.gridstackNode.w;
+                    widget.order_column = item.gridstackNode.x;
+                    widget.order_row = item.gridstackNode.y;
+                    newSnapshot.push(widget);
+                }
+            });
+            // sync properties
+            await $wire.syncWidgets(newSnapshot);
```

Comment:
Can't this be done in one livewire method?

Reply:
you are calling 3 livewire methods, one after the other, why not call one method that does all three? or at least without await before each call

Reply:
i can edit saveDashboard to accept new snapsot - to avoid 2 rounds to server.
i have to await it to avoid possible racing conditions